### ISSUE TITLE
Passing arguments to ruby-build

### ISSUE THREAD

**Issue by ipatch** [asdf-vm/asdf-ruby#75]
**TL;DR**

I'd like to keep the source directory and the tarball file [ruby-build](https://github.com/rbenv/ruby-build) downloads when running,

```
asdf install ruby 2.5.1
```

Scanning through the [README.md](https://github.com/rbenv/ruby-build/blob/master/README.md) I came across how build environment variables can be passed to **ruby-build** however, off the top of my head, I couldn't see or understand how I could pass an argument or flag to **ruby-build** more specifically the, the `-k` or `--keep` which would allow me to keep the tarball and extracted source directory of the tarball when running the above ☝️mentioned **asdf** command.

Didn't really see how a command flag could be passed to the the **ruby-build** command, so I'm not sure if I should be pointing the finger here, or pointing the finger at [**ruby-build**](https://github.com/rbenv/ruby-build#custom-build-configuration). 

It'd sure be swell, 🌈🦄if asdf supported passing flags to the tools that are building the languages, ie. the scenario I am describing, however I can also see how it could be a quick fix for the ruby-build guys to implement another **custom build configuration** variable for this situation, but I think [**this**](https://github.com/asdf-vm/asdf-ruby/blob/master/lib/utils.sh#L41-L42) would still be a glaring issue even if a **custom build configuration** was added for keeping the downloaded tarbabll and source directory with **ruby-build**, thus bringing this conversation back to the original thought, it sure would be nice if asdf could or would support passing flags to the build tools that the plugins rely on.

<img width="922" alt="ruby-build-custom-build-configuration-asdf-gh-issue" src="https://user-images.githubusercontent.com/613805/45319608-da7eb100-b505-11e8-9464-bfc6eac115a5.png">

**Comment by Stratus3D** [comment]
Yes, this needs to be addressed. We will need to add support for keeping source to asdf itself, because it currently removes the entire build directory as you pointed out. Follow https://github.com/asdf-vm/asdf/issues/74 for updates.
